tez-api/src/main/java/org/apache/tez/client/TezClientUtils.java

```diff
@@ -820,6 +820,23 @@ public static String addDefaultsToTaskLaunchCmdOpts(String vOpts, Configuration
     return vOpts;
   }
 
+  private static String getDefaultAmLaunchOpts() {
+    return getJavaVersion() >= 9 ? TezConfiguration.TEZ_AM_LAUNCH_CMD_OPTS_JDK9_DEFAULT
+      : TezConfiguration.TEZ_AM_LAUNCH_CMD_OPTS_DEFAULT;
+  }
+
+  private static String getDefaultTaskLaunchOpts() {
+    return getJavaVersion() >= 9 ? TezConfiguration.TEZ_TASK_LAUNCH_CMD_OPTS_JDK9_DEFAULT
+      : TezConfiguration.TEZ_TASK_LAUNCH_CMD_OPTS_DEFAULT;
+  }
+
+  private static int getJavaVersion() {
+    String javaVersionString = System.getProperty("java.version");
```

Comment:
I originally detected the presence of java.lang.runtime.Version class to avoid parsing the version string. Knowing the actual version could be more useful in the future though so let's try to keep this for now.